modules/twinklespeedy.js

```diff
@@ -626,12 +626,20 @@ Twinkle.speedy.fileList = [
 		label: 'F9: Unambiguous copyright infringement',
 		value: 'imgcopyvio',
 		tooltip: 'The file was copied from a website or other source that does not have a license compatible with Wikipedia, and the uploader neither claims fair use nor makes a credible assertion of permission of free use. Sources that do not have a license compatible with Wikipedia include stock photo libraries such as Getty Images or Corbis. Non-blatant copyright infringements should be discussed at Wikipedia:Files for deletion',
-		subgroup: {
-			name: 'imgcopyvio_url',
-			type: 'input',
-			label: 'URL of the copyvio, including the "http://".  If you cannot provide a URL, please do not use CSD F9.  (Exception: for copyvios of non-Internet sources, leave the box blank.) ',
-			size: 60
-		}
+		subgroup: [
+			{
+				name: 'imgcopyvio_url',
+				type: 'input',
+				label: 'URL of the copyvio, including the "http://".  If the copyvio is of a non-internet source and you cannot provide a URL, you must use the deletion rationale box. ',
+				size: 60
+			},
+			{
+				name: 'imgcopyvio_reason',
```

Comment:
One thing I'd like some feedback on is whether this is a good idea, or whether it'd be better served as a popup.  Having two boxes takes up a lot of visual space, and the url message is necessarily already very long and awkward looking.  In particular I worry about the "tag multiple" option, where this will especially stand out.

Three options could be:
1. Previous status quo: Don't include the `reason` parameter for F9, keep `url` optional but heavily encouraged
2. This commit: Have both `url` and `reason` inputs, require at least one of them
3. Alt: Have `url` input box, provide a `prompt` for the `reason` parameter if they leave `url` blank.

Reply:
I've written this as option 2, but I'm leaning toward doing option 3, for overall clarity and to encourage people not to use the reason/rationale input willy-nilly

Reply:
2 or 3 should be OK. I'm leaning toward 3. It should still require `rationale` in order to tag if `url` wasn't specified.

Reply:
2>3>1. In cases where I'm contradicting a statement by the uploader other than {{own work}}, I'd like to leave a reason along with the URL. I'm not so concerned about the two-boxes-being-bigger-than-one-box issue as they're only shown when someone has already clicked the radio button.

Reply:
@AntiCompositeNumber 	It's more that the text for url wraps likely around, so there's two lines plus a big box for that, then another line for rationale+text.  It's awkward but certainly manageable.  I think the chief advantage of option 3 is further encouragement *not* to use `rationale`, but then again we have that long-ass sentence so it should be pretty clear.  Besides, the awkward placement would surely draw attention!

Maybe the thing to do is go with 3 and if people complain can do 2.

Reply:
Fixed my above comment fwiw, I meant to suggest sticking with option 2 (this current PR) and going to 3 if folks complain.  I do suppose being able input *both* `url` and `rationale` could be productive.